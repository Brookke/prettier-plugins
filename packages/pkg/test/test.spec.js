const { join } = require('path')

const prettier = require('prettier')

const pkg = require('../package.json')

const shuffle = arr => {
  const result = arr.slice()
  for (let i = result.length - 1; i > 0; i--) {
    const rand = Math.floor(Math.random() * (i + 1))
    ;[result[i], result[rand]] = [result[rand], result[i]]
  }
  return result
}

test('randomize', () => {
  const options = {
    filepath: join(__dirname, 'package.json'),
    parser: 'json-stringify',
    plugins: ['prettier-plugin-pkg'],
  }
  const keys = shuffle(Object.keys(pkg))
  const fixture = {}

  for (const key of keys) {
    fixture[key] = pkg[key]
  }

  const input = JSON.stringify(fixture, null, 2)
  expect(prettier.format(input, options)).toMatchSnapshot()
})

test('preprocess', () => {
  const options = {
    filepath: join('package.json'),
    parser: 'json-stringify',
    plugins: ['prettier-plugin-pkg'],
    preprocess: input => {
      const { version, repository } = JSON.parse(input)
      const result = { repository, version }
      return result
    },
  }
  const keys = shuffle(Object.keys(pkg))
  const fixture = {}

  for (const key of keys) {
    fixture[key] = pkg[key]
  }

  const input = JSON.stringify(fixture, null, 2)
  const output = prettier.format(input, options)

  expect(output).toMatchSnapshot()
})

test('not package.json', () => {
  const options = {
    filepath: 'batman.json',
    parser: 'json-stringify',
    plugins: ['prettier-plugin-pkg'],
  }
  const fixture = { version: 'batman', name: 'joker' }
  const input = JSON.stringify(fixture, null, 2)
  const output = prettier.format(input, options)

  expect(input.trim()).toBe(output.trim())
})

test('broken json', () => {
  const options = {
    filepath: 'broken.json',
    parser: 'json-stringify',
    plugins: ['prettier-plugin-pkg'],
  }

  const broken = `{
  "name": "prettier-plugin-pkg",
  "batmam": {]
}`

  expect(() => prettier.format(broken, options)).toThrow()
})
