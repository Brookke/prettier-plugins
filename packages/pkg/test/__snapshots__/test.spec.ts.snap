// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`preprocess 1`] = `
"{
  \\"name\\": \\"@rxts/prettier-plugin-pkg\\",
  \\"version\\": \\"0.0.0\\",
  \\"description\\": \\"An opinionated package.json formatter plugin for Prettier\\",
  \\"repository\\": \\"git@github.com/rx-ts/prettier.git\\",
  \\"homepage\\": \\"https://github.com/rx-ts/prettier/blob/master/packages/pkg\\",
  \\"author\\": \\"JounQin <admin@1stg.me>\\",
  \\"license\\": \\"MPL-2.0\\",
  \\"engines\\": {
    \\"node\\": \\">= 8.0.0\\"
  },
  \\"main\\": \\"lib/cjs\\",
  \\"module\\": \\"lib/esm\\",
  \\"es2015\\": \\"lib/es2015\\",
  \\"files\\": [
    \\"lib\\",
    \\"rules\\",
    \\"LICENSE\\",
    \\"README.md\\"
  ],
  \\"keywords\\": [
    \\"package\\",
    \\"package.json\\",
    \\"pkg\\",
    \\"plugin\\",
    \\"prettier\\",
    \\"prettier-plugin\\"
  ],
  \\"scripts\\": {
    \\"bootstrap\\": \\"yarn build && tsc --module commonjs --outDir lib/cjs\\",
    \\"build\\": \\"ts-node -O '{\\\\\\"module\\\\\\":\\\\\\"commonjs\\\\\\"}' build\\"
  },
  \\"peerDependencies\\": {
    \\"prettier\\": \\"^1.18.2\\"
  }
}
"
`;

exports[`randomize 1`] = `
"{
  \\"name\\": \\"@rxts/prettier-plugin-pkg\\",
  \\"version\\": \\"0.0.0\\",
  \\"description\\": \\"An opinionated package.json formatter plugin for Prettier\\",
  \\"repository\\": \\"git@github.com/rx-ts/prettier.git\\",
  \\"homepage\\": \\"https://github.com/rx-ts/prettier/blob/master/packages/pkg\\",
  \\"author\\": \\"JounQin <admin@1stg.me>\\",
  \\"license\\": \\"MPL-2.0\\",
  \\"engines\\": {
    \\"node\\": \\">= 8.0.0\\"
  },
  \\"main\\": \\"lib/cjs\\",
  \\"module\\": \\"lib/esm\\",
  \\"es2015\\": \\"lib/es2015\\",
  \\"files\\": [
    \\"lib\\",
    \\"rules\\"
  ],
  \\"keywords\\": [
    \\"package\\",
    \\"package.json\\",
    \\"pkg\\",
    \\"plugin\\",
    \\"prettier\\",
    \\"prettier-plugin\\"
  ],
  \\"peerDependencies\\": {
    \\"prettier\\": \\"^1.18.2\\"
  },
  \\"commitlint\\": {
    \\"extends\\": [
      \\"@1stg\\"
    ]
  },
  \\"eslintIgnore\\": [
    \\"CHANGELOG.md\\",
    \\"lib\\",
    \\"!.*.js\\"
  ]
}
"
`;
